pkgs/stdenv/generic/make-derivation.nix

```
@@ -62,7 +62,7 @@ let
                 f0 self super
               else x;
         in
-          makeDerivationExtensible (self: attrs // f self attrs))
+          makeDerivationExtensible (self: attrs // (if builtins.isFunction f0 || f0?__functor then f self attrs else f0)))
```

does this work the same?

```suggestion
          makeDerivationExtensible (self: attrs // (if lib.isFunction f0 then f self attrs else f0)))
```

---

lib is avoided in these 2 functions for performance reasons